### PR TITLE
Seabeta 558 remove personal information

### PR DESCRIPTION
**Pull request by DanCorderSoftwire**
Required refactoring the question flow as we don't always have the right information to go back to the page before.
Also fixed bugs when editing property type and age

### CODE REVIEW

File: SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs [hunk 1]
```diff
@@ -249,72 +244,98 @@ public async Task<IActionResult> AskForPostcode_Post(AskForPostcodeViewModel vie
...
+                forwardArgs.Controller,
+                // TODO: seabeta-579 we'd rather use the values from the question flow service
+                new { reference = viewModel.Reference, postcode = viewModel.Postcode, number = viewModel.HouseNameOrNumber });
         }
 
         
-        [HttpGet("address/{reference}")]
-        public async Task<ViewResult> ConfirmAddress_Get(string reference)
+        [HttpGet("address/{reference}/{postcode}/{number}")]
+        public async Task<ViewResult> ConfirmAddress_Get(string reference, string postcode, string number)
```
Comment (CodrinCotarlanSoftwire) [inline SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs:260]: any reason for renaming this from `houseNameOrNumber` to `number`?

Comment (DanCorderSoftwire) [reply SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs:260]: Not really, it was just shorter

File: SeaPublicWebsite.UnitTests/QuestionFlowServiceTests.cs [hunk 2]
```diff
@@ -148,85 +148,94 @@ public void RunSkipLinkTestCases(QuestionFlowServiceTestCase testCase)
...
             ),
             new PathByActionArguments(
-                nameof(EnergyEfficiencyController.NoEpcFound_Get),
+                nameof(EnergyEfficiencyController.AskForPostcode_Get),
                 "EnergyEfficiency",
                 new { reference = "ABCDEFGH" }
             )),
         new(
-            "Property type goes back to confirm address if address confirmed but details confirmed is null",
+            "Property type goes back to no EPC found if searched for EPC but no EPC set",
```
Comment (natdeanlewissoftwire) [inline SeaPublicWebsite.UnitTests/QuestionFlowServiceTests.cs:221]: does this cover the case that a user clicked "Yes" but then clicked "cancel and continue without epc? this should go back to FindEpc not to NoEpcFound I think.

**Review by natdeanlewissoftwire — APPROVED** [review]

File: SeaPublicWebsite.UnitTests/QuestionFlowServiceTests.cs (hunk 2 above)
Comment (DanCorderSoftwire) [reply SeaPublicWebsite.UnitTests/QuestionFlowServiceTests.cs:221]: Good spot, this is broken

File: SeaPublicWebsite/Views/EnergyEfficiency/ConfirmSingleAddress.cshtml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                        This certificate may be registered to your property or one of the properties nearby that shares part of your address.
+                    </p>
+                </text>
+
+                @await Html.GovUkNotificationBanner(new NotificationBannerViewModel
+                {
+                    Type = "null",
+                    TitleHtml = @<text>
+                                    <div>Certificate</div>
+                                 </text>,
```
Comment (CodrinCotarlanSoftwire) [inline SeaPublicWebsite/Views/EnergyEfficiency/ConfirmSingleAddress.cshtml:44]: no need for <text>

**Review by CodrinCotarlanSoftwire — APPROVED** [review]
No issues that I could find :) just small comments

**Pull request merged**